src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -106,6 +125,25 @@ public AssetProvider CreateAssetProvider(Checksum solutionChecksum, SolutionAsse
             bool updatePrimaryBranch,
             Func<Solution, ValueTask<T>> implementation,
             CancellationToken cancellationToken)
+        {
+            try
+            {
+                var pinnedSolution = await GetPinnedSolutionAsync(assetProvider, solutionChecksum, workspaceVersion, updatePrimaryBranch, cancellationToken).ConfigureAwait(false);
+                await using (pinnedSolution.ConfigureAwait(false))
```

Comment:
@stephentoub the need for the two `.ConfigureAwaits` seems... really unfortunate.  It's also scary to see the acquisition of the IAsyncDisposable, and then a separate statement before it it actually placed into the using.

Is there a better way to do this?